metadata-ingestion/src/datahub/ingestion/source/aerospike.py

```diff
@@ -0,0 +1,591 @@
+            )
+            PYTHON_TYPE_TO_AEROSPIKE_TYPE[field_type] = "unknown"
+            type_string = "unknown"
+
+        return type_string
+
+    def get_field_type(
+        self, field_type: Union[Type, str], set_name: str
+    ) -> SchemaFieldDataType:
+        """
+        Maps types encountered in Aerospike Python client to corresponding schema types.
+
+        Parameters
+        ----------
+            field_type:
+                type of Python object
+            set_name:
+                name of set (for logging)
+        """
+        TypeClass: Optional[Type] = _field_type_mapping.get(field_type)
+
+        if TypeClass is None:
+            self.report.report_warning(
+                set_name, f"unable to map type {field_type} to metadata schema"
+            )
+            TypeClass = NullTypeClass
+
+        return SchemaFieldDataType(type=TypeClass())
+
+    def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
+        sets_info: str = self.aerospike_client.info_random_node("sets")
+        sets_info = (
+            sets_info[len("sets\t") :] if sets_info.startswith("sets\t") else sets_info
+        )
+        sets_info = sets_info[: -len(";\n")] if sets_info.endswith(";\n") else sets_info
+
+        all_sets: List[AerospikeSet] = [
+            AerospikeSet(item) for item in sets_info.split(";") if item
+        ]
+        if self.config.ignore_empty_sets:
+            all_sets = [
+                aerospike_set for aerospike_set in all_sets if aerospike_set.objects > 0
+            ]
+
+        namespaces = list(set([aerospike_set.ns for aerospike_set in all_sets]))
+
+        # traverse namespaces in sorted order so output is consistent
+        for namespace in sorted(namespaces):
+            if namespace in DENY_NAMESPACE_LIST:
+                continue
+            if not self.config.namespace_pattern.allowed(namespace):
+                self.report.report_dropped(namespace)
+                continue
+
+            namespace_key = NamespaceKey(
+                namespace=namespace,
+                platform=self.platform,
+                instance=self.config.platform_instance,
+                env=self.config.env,
+            )
+            yield from gen_containers(
+                container_key=namespace_key,
+                name=namespace,
+                sub_types=[DatasetContainerSubTypes.NAMESPACE],
+            )
+
+            ns_sets: List[AerospikeSet] = [
+                aerospike_set
+                for aerospike_set in all_sets
+                if aerospike_set.ns == namespace
+            ]
+            if self.config.include_xdr:
+                xdr_sets = self.xdr_sets(
+                    namespace, [aerospike_set.set for aerospike_set in ns_sets]
+                )
+            else:
+                xdr_sets = {}
+
+            # traverse sets in sorted order so output is consistent
+            for curr_set in sorted(ns_sets, key=lambda x: x.set):
```

Comment:
This method is getting a bit long / hard to read.. consider breaking down into multiple sub-methods